### PR TITLE
WEBAPP-145: Test and refactor route events

### PR DESCRIPTION
**Pull request by klinzo**

### CODE REVIEW

File: src/modules/endpoint/models/LanguageModel.js [hunk 1]
```diff
@@ -1,15 +1,15 @@
 class LanguageModel {
   constructor (code, name) {
-    this._name = code
-    this._code = name
+    this._code = code
+    this._name = name
```
Comment (maxammann) [inline src/modules/endpoint/models/LanguageModel.js]: 👍

File: src/routes/events/containers/EventsPage.js [hunk 2]
```diff
@@ -6,41 +6,85 @@ import Spinner from 'react-spinkit'
...
+  }
+
+  // we must not call dispatch in componentWillUpdate or componentDidUpdate
+  componentWillReceiveProps (nextProps) {
+    if (nextProps.events === this.props.events && nextProps.path === this.props.path) {
+      // no relevant prop changes
+      return
+    }
 
+    if (nextProps.path) {
```
Comment (maxammann) [inline src/routes/events/containers/EventsPage.js]: If you have only one id in the path and no hierarchy of paths, you can change the route definitoin in routes/index.js to '/events/:id'. Then you can access the id by state.router.params.id.

No more need for path parsing :)

File: src/routes/events/containers/EventsPage.js [hunk 3]
```diff
@@ -6,41 +6,85 @@ import Spinner from 'react-spinkit'
...
+      // all events
+      this.props.dispatchLanguageChangeUrls(this.mapLanguageToUrl, nextProps.languages)
+    }
+  }
+
+  render () {
     if (this.props.path) {
       // event with the given id from this.props.path
-      const event = events.find((event) => event.event.id.toString() === this.props.path.replace('/', ''))
+      const event = this.props.events.find((event) => event.id.toString() === this.props.path.replace('/', ''))
```
Comment (maxammann) [inline src/routes/events/containers/EventsPage.js]: You have this code 3 times in this file: `this.props.events.find((event) => event.id.toString() === this.props.path.replace('/', ''))`
Better extract it to a method

File: src/routes/events/containers/__tests__/EventsPage.spec.js [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+  const mockStore = configureMockStore([thunk])
+
+  test('should map state to props', () => {
+    const store = mockStore({
+      endpoint: new Payload(false),
+      endpoint1: new Payload(false),
+      router: {params: {location: 'augsburg', language: 'en', _: ''}}
+    })
+
+    const wrapper = mount(
```
Comment (maxammann) [inline src/routes/events/containers/__tests__/EventsPage.spec.js]: Expect is missing here. (I think this is still to do :))

File: src/routes/search/containers/SearchPage.js [hunk 5]
```diff
@@ -64,22 +73,21 @@ class SearchPage extends React.Component {
...
       </div>
     )
   }
 }
 
 const mapStateToProps = (state) => ({
+  languages: state.languages,
   language: state.router.params.language,
   location: state.router.params.location,
   path: state.router.params['_'] // _ contains all the values from *
```
Comment (maxammann) [inline src/routes/search/containers/SearchPage.js]: I think this is not needed in SearchPage

File: src/routes/events/containers/EventsPage.js (hunk 2 above)
Comment (klinzo) [reply src/routes/events/containers/EventsPage.js]: nice

File: src/routes/events/containers/EventsPage.js (hunk 3 above)
Comment (klinzo) [reply src/routes/events/containers/EventsPage.js]: done

File: src/routes/events/containers/__tests__/EventsPage.spec.js (hunk 4 above)
Comment (klinzo) [reply src/routes/events/containers/__tests__/EventsPage.spec.js]: yeah right, you asked me to push this :P

File: src/routes/events/containers/__tests__/EventsPage.spec.js [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+      const eventsPageProps = tree.find(ConnectedEventsPage).childAt(0).props()
+
+      let countActions = store.getActions().length
+
+      eventsPageProps.setLanguageChangeUrls(mapLanguageToUrl, languages, availableLanguages)
+      expect(store.getActions()).toHaveLength(countActions + 1)
+
+      expect(store.getActions()).toContainEqual({
+        payload: testUrls,
+        type: 'SET_LANGUAGE_CHANGE_URLS'
```
Comment (maxammann) [inline src/routes/events/containers/__tests__/EventsPage.spec.js]: 👍 for the name 'SET_LANGUAGE_CHANGE_URLS' :D

File: src/routes/events/containers/__tests__/EventsPage.spec.js [hunk 7]
```diff
@@ -0,0 +1,255 @@
...
+      const tree = mount(
+        <Provider store={store}>
+          <ConnectedEventsPage />
+        </Provider>
+      )
+
+      // todo expect setLanguageChangeUrls action to be in store, but as we don't get events and languages from our
+      // mocked endpoint no action is dispatched
+
+      const eventsPageProps = tree.find(ConnectedEventsPage).childAt(0).props()
```
Comment (maxammann) [inline src/routes/events/containers/__tests__/EventsPage.spec.js]: We should finish the testing for the passing of the props in this task. The next tasks will need a working solution.

Did you already try the solution with the promises and real store?

Comment (klinzo) [reply src/routes/events/containers/__tests__/EventsPage.spec.js]: yeah I tried but couldn't get it to work

Comment (klinzo) [reply src/routes/events/containers/__tests__/EventsPage.spec.js]: created a task

File: src/routes/events/components/EventList.js [hunk 8]
```diff
@@ -1,87 +1,48 @@
...
-                   event={event.event}
-                   parentUrl={this.props.url}
-                   thumbnailPlaceholder={event.thumbnailPlaceholder}
-                   language={this.props.language}
-                   isFirst={index === 0}/>)
-          : <div className={style.noEvents}>{t('currentlyNoEvents')}</div>
-        }
+        <Caption title={t('news')} />
+        {elements}
       </div>
```
Comment (Schedulaar) [inline src/routes/events/components/EventList.js]: Maybe you can replace this by 
```javascript
return (
  <React.Fragment>
     <Caption title={t('news')] />
     <div className={style.list}>
       {elements}
     </div>
  </React.Fragment>
)
```
then you can replace `.first-event { border-top: 2px solid var(--theme-color); }` by `.list { border-bottom: 2px solid var(--theme-color); }` in the stylesheet. That would make more sense. Then you can also remove the `isFirst` logic from the `<Event />`
These are things we would now need to test, so simplifying is in your interest :)

**Pull request merged**